patchwork/steps/ModifyCode/ModifyCode.py

```diff
@@ -81,16 +96,78 @@ def run(self) -> dict:
             return dict(modified_code_files=[])
 
         for code_snippet, extracted_response in sorted_list:
-            uri = code_snippet.get("uri")
+            # Use Path for consistent path handling
+            file_path = Path(code_snippet.get("uri", ""))
             start_line = code_snippet.get("startLine")
             end_line = code_snippet.get("endLine")
             new_code = extracted_response.get("patch")
 
             if new_code is None:
                 continue
 
-            replace_code_in_file(uri, start_line, end_line, new_code)
-            modified_code_file = dict(path=uri, start_line=start_line, end_line=end_line, **extracted_response)
+            # Get the original content for diffing
+            diff = ""
+            
+            if file_path.exists():
+                try:
+                    # Create a temporary directory with restricted permissions
+                    with tempfile.TemporaryDirectory(prefix='modifycode_') as temp_dir:
```

Comment:
Instead of creating files, can we diff the contents of the files in memory instead?